pkg/operator/vtgate/deployment.go

```diff
@@ -122,8 +123,12 @@ func UpdateDeployment(obj *appsv1.Deployment, spec *Spec) {
 	update.Labels(&obj.Labels, spec.Labels)
 	// Tell Deployment to set the same labels on the Pods it creates.
 	update.Labels(&obj.Spec.Template.Labels, spec.Labels)
-
-	update.Annotations(&obj.Spec.Template.Annotations, spec.PodAnnotations)
+	// Clobber existing annotations so that we don't mutate the input map.
```

Comment:
The existing `spec.PodAnnotations` was already made for this purpose, so we can just reuse that. Let's rename it to `spec.Annotations` to be consistent with all our other new spec fields, though. Then this just needs to be one line like for vtctld:

```go
obj.Spec.Template.Annotations = spec.Annotations
```

Reply:
Done.